runtimes/c/runtime.h

```diff
@@ -0,0 +1,324 @@
+                          CATALA_DURATION x1,
+                          CATALA_DURATION x2);
+
+CATALA_BOOL o_eq_boo_boo (CATALA_BOOL x1, CATALA_BOOL x2);
+
+CATALA_BOOL o_eq_int_int (CATALA_INT x1, CATALA_INT x2);
+
+CATALA_BOOL o_eq_rat_rat (CATALA_DEC x1, CATALA_DEC x2);
+
+CATALA_BOOL o_eq_mon_mon (CATALA_MONEY x1, CATALA_MONEY x2);
+
+CATALA_BOOL o_eq_dat_dat (CATALA_DATE x1, CATALA_DATE x2);
+
+CATALA_BOOL o_eq_dur_dur (const catala_code_position* pos,
+                          CATALA_DURATION x1, CATALA_DURATION x2);
+
+CATALA_BOOL o_lt_int_int (CATALA_INT x1, CATALA_INT x2);
+
+CATALA_BOOL o_lt_rat_rat (CATALA_DEC x1, CATALA_DEC x2);
+
+CATALA_BOOL o_lt_mon_mon (CATALA_MONEY x1, CATALA_MONEY x2);
+
+CATALA_BOOL o_lt_dat_dat (CATALA_DATE x1, CATALA_DATE x2);
+
+CATALA_BOOL o_lt_dur_dur (const catala_code_position* pos,
+                          CATALA_DURATION x1, CATALA_DURATION x2);
+
+CATALA_BOOL o_lte_int_int (CATALA_INT x1, CATALA_INT x2);
+
+CATALA_BOOL o_lte_rat_rat (CATALA_DEC x1, CATALA_DEC x2);
+
+CATALA_BOOL o_lte_mon_mon (CATALA_MONEY x1, CATALA_MONEY x2);
+
+CATALA_BOOL o_lte_dat_dat (CATALA_DATE x1, CATALA_DATE x2);
+
+CATALA_BOOL o_lte_dur_dur (const catala_code_position* pos,
+                           CATALA_DURATION x1, CATALA_DURATION x2);
+
+CATALA_BOOL o_gt_int_int (CATALA_INT x1, CATALA_INT x2);
+
+CATALA_BOOL o_gt_rat_rat (CATALA_DEC x1, CATALA_DEC x2);
+
+CATALA_BOOL o_gt_mon_mon (CATALA_MONEY x1, CATALA_MONEY x2);
+
+CATALA_BOOL o_gt_dat_dat (CATALA_DATE x1, CATALA_DATE x2);
+
+CATALA_BOOL o_gt_dur_dur (const catala_code_position* pos,
+                          CATALA_DURATION x1, CATALA_DURATION x2);
+
+CATALA_BOOL o_gte_int_int (CATALA_INT x1, CATALA_INT x2);
+
+CATALA_BOOL o_gte_rat_rat (CATALA_DEC x1, CATALA_DEC x2);
+
+CATALA_BOOL o_gte_mon_mon (CATALA_MONEY x1, CATALA_MONEY x2);
+
+CATALA_BOOL o_gte_dat_dat (CATALA_DATE x1, CATALA_DATE x2);
+
+CATALA_BOOL o_gte_dur_dur (const catala_code_position* pos,
+                           CATALA_DURATION x1, CATALA_DURATION x2);
+
+const CATALA_ARRAY(X) o_filter (catala_closure* cls, const CATALA_ARRAY(X) x);
+
+const CATALA_ARRAY(Y) o_map (catala_closure* cls, const CATALA_ARRAY(X) x);
+
+const void* o_fold (catala_closure* cls,
+                    const void* init, const CATALA_ARRAY(X) x);
+
+const void* o_reduce (catala_closure* cls,
+                      const void* dft, const CATALA_ARRAY(X) x);
+
+const CATALA_ARRAY(Z) o_map2 (const catala_code_position* pos,
+                        catala_closure* cls,
+                        const CATALA_ARRAY(X) x,
+                        const CATALA_ARRAY(Y) y);
+
+const CATALA_ARRAY(Z) o_concat (const CATALA_ARRAY(X) x,
+                                const CATALA_ARRAY(Y) y);
+
+const CATALA_OPTION() handle_exceptions
+  (const CATALA_ARRAY(const catala_code_position*) pos,
+   const CATALA_ARRAY(const CATALA_OPTION()) e);
+
+/* --- Runtime initialisation --- */
+
+void catala_init();
```

Comment:
Comment here or `README` with an example of how to call the C runtime properly from the client perspective?